workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/Templates/UnityWorkerMenuGenerator.tt

```diff
@@ -62,21 +62,94 @@ var workerType = workerTypeList[i];
+
+        private static bool ValidateWorkersCanBuildForEnvironment(BuildEnvironment environment, string[] workerTypes)
+        {
+            var canInitiateBuild = true;
+            var spatialOSBuildConfiguration = SpatialOSBuildConfiguration.GetInstance();
+
+            foreach (var workerType in workerTypes)
+            {
+                try
+                {
+                    WorkerBuilder.AssertWorkerCanBuildForEnvironment(workerType, environment);
+                }
+                catch (System.Exception e)
+                {
+                    canInitiateBuild = false;
+
+                    Debug.LogException(e, spatialOSBuildConfiguration);
+                }
+            }
+
+            return canInitiateBuild;
+        }
+
+        private static void MenuBuild(BuildEnvironment environment, params string[] workerTypes)
         {
             // Delaying build by a frame to ensure the editor has re-rendered the UI to avoid odd glitches.
             EditorApplication.delayCall += () =>
             {
-                LocalLaunch.BuildConfig();
+                if (!ValidateWorkersCanBuildForEnvironment(environment, workerTypes))
+                {
+                    DisplayBuildFailureDialog();
+
+                    return;
+                }
 
-                foreach (var platform in platforms)
+                try
                 {
-                    WorkerBuilder.BuildWorkerForEnvironment(platform, environment);
+                    LocalLaunch.BuildConfig();
+
+                    foreach (var workerType in workerTypes)
+                    {
+                        WorkerBuilder.BuildWorkerForEnvironment(workerType, environment);
+                    }
+
+                    Debug.LogFormat("Completed build for {0} target", environment);
                 }
+                catch (System.Exception)
+                {
+                    DisplayBuildFailureDialog();
 
-                Debug.LogFormat("Completed build for {0} target", environment);
+                    throw;
+                }
             };
         }
+
+        private static void DisplayBuildFailureDialog()
+        {
+            EditorUtility.DisplayDialog("Build Failed",
+                "Build failed. Please see the Unity Console Window for information.",
+                "OK");
+        }
+
+        [MenuItem(EditorConfig.ParentMenu + "/Validate Build Support", false, EditorConfig.MenuOffset + <#= workerTypeList.Count + 1 #>)]
+        private static void ValidateBuildSupport()
+        {
+            Debug.Log("Checking for build platform support...");
+
+            var validationsPassed = true;
+
+            Debug.Log("Validating local builds");
+            if (!ValidateWorkersCanBuildForEnvironment(BuildEnvironment.Local, AllWorkers))
+            {
+                validationsPassed = false;
+            }
+
+            Debug.Log("Validating cloud builds");
```

Comment:
Do we need all of these debug logs here?

Reply:
It helps to see validation status IMO, but can merge it into one multi-line log if that's preferred

Reply:
I think the just the last one is fine - either validation passes and it logs or you get the popup